cgi/org.pl

```diff
@@ -358,18 +358,41 @@
 	if ($type eq "edit") {
 
 		store_org($org_ref);
+		$template_data_ref->{result} = lang("edit_org_result");
 	}
 	elsif ($type eq 'user_delete') {
 
 		if (is_user_in_org_group($org_ref, $User_id, "admins")) {
 			remove_user_by_org_admin($orgid, single_param('user_id'));
+			$template_data_ref->{result} = lang("edit_org_result");
 		}
 		else {
 			display_error_and_exit($Lang{error_no_permission}{$lang}, 403);
 		}
 
 	}
-	$template_data_ref->{result} = lang("edit_org_result");
+	elsif ($type eq 'add_users') {
+		if (is_user_in_org_group($org_ref, $User_id, "admins")) {
+			my $email_list = remove_tags_and_quote(single_param('email_list'));
+			my $email_ref = add_users_to_org_by_admin($orgid, $email_list);
+
+			my @messages;
```

Comment:
Looks good. But all the logic to create the messages with the lists of users invited and added should be moved to the HTML template file, so that we don't have HTML tags in the Perl code, and that designers can easily change the format of the messages (for instance using a <ul> list to list the users invited / added).